pkgs/development/python-modules/manticore/default.nix

```diff
@@ -38,16 +40,21 @@ buildPythonPackage rec {
     crytic-compile
     wasm
     psutil
-    z3-solver
     setuptools
     capstone
     pyelftools
     unicorn
   ];
 
-  patchPhase = ''sed -ie s/pyevmasm==0.2.0/pyevmasm/ setup.py'';
+  postPatch = ''
+    sed -ie s/pyevmasm==0.2.0/pyevmasm/ setup.py
+    # Python API is not used in the code, only z3 from PATH
+    sed -ie s/z3-solver// setup.py
+    '';
+
+  makeWrapperArgs = [ ''--prefix PATH : "${z3}/bin"'' ];
```

Comment:
if you want this as a python module, you can't rely on wrapping. Another package may import this and won't have z3 on PATH. You have to do a patch so that the logic which calls the command is passed